### PR TITLE
Add 'google_organization' data source

### PR DESCRIPTION
**Pull request by ewbankkit**
Fixes https://github.com/terraform-providers/terraform-provider-google/issues/886.

### CODE REVIEW

**Comment by ewbankkit** [comment]
```
make testacc TEST=./google/ TESTARGS='-run=TestAccDataSourceGoogleOrganization_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google/ -v -run=TestAccDataSourceGoogleOrganization_ -timeout 120m
=== RUN   TestAccDataSourceGoogleOrganization_basic
--- PASS: TestAccDataSourceGoogleOrganization_basic (0.97s)
PASS
ok  	github.com/terraform-providers/terraform-provider-google/google	0.974s
```

File: google/data_source_google_organization.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+		organization = resp
+	}
+
+	parts := strings.Split(organization.Name, "/")
+	if len(parts) != 2 {
+		return fmt.Errorf("Invalid organization name. Expecting organizations/{organization_id}")
+	}
+
+	d.SetId(parts[1])
```
Comment (rosbo) [inline google/data_source_google_organization.go]: You can use: https://github.com/terraform-providers/terraform-provider-google/blob/master/google/self_link_helpers.go#L68

File: google/resourcemanager_helpers.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package google
+
+import (
+	"fmt"
+
+	"google.golang.org/api/cloudresourcemanager/v1"
+)
+
+func getResourceName(resourceId *cloudresourcemanager.ResourceId) string {
```
Comment (rosbo) [inline google/resourcemanager_helpers.go]: I don't see any usage for this method?

File: google/data_source_google_organization.go (hunk 1 above)
Comment (ewbankkit) [reply google/data_source_google_organization.go]: Changed as suggested.

File: google/resourcemanager_helpers.go (hunk 2 above)
Comment (ewbankkit) [reply google/resourcemanager_helpers.go]: Hmm, this code must have come from something else I was noodling on. Removed those 2 source files.

**Comment by ewbankkit** [comment]
I'm going to make some additional changes based on review comments in https://github.com/terraform-providers/terraform-provider-google/pull/889 - e.g. use `ConflictsWith` in the schema.

**Pull request merged**